core/bot_framework_adapter_test.go

```diff
@@ -20,38 +20,56 @@
 package core_test
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
-	"log"
-	"os"
+	"net/http"
+	"net/http/httptest"
+	"testing"
 
+	"github.com/infracloudio/msbotbuilder-go/connector/auth"
+	"github.com/infracloudio/msbotbuilder-go/connector/client"
 	"github.com/infracloudio/msbotbuilder-go/core"
 	"github.com/infracloudio/msbotbuilder-go/core/activity"
 	"github.com/infracloudio/msbotbuilder-go/schema"
+
+	"github.com/stretchr/testify/assert"
 )
 
-func Example() {
+func serverMock() *httptest.Server {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/v3/conversations/abcd1234/activities", msTeamsMockMock)
 
-	// Load settings from environment variables to AdapterSetting.
-	setting := core.AdapterSetting{
-		AppID:       os.Getenv("APP_ID"),
-		AppPassword: os.Getenv("APP_PASSWORD"),
-	}
+	srv := httptest.NewServer(handler)
 
-	// Make an adapter to perform operations with the Bot Framework using this library.
-	adapter, err := core.NewBotAdapter(setting)
-	if err != nil {
-		log.Fatal(err)
-	}
+	return srv
+}
 
-	// Create a handler that defines operations to be performed on respective events.
-	// Following defines the operation to be performed on the 'message' event.
-	var customHandler = activity.HandlerFuncs{
-		OnMessageFunc: func(turn *activity.TurnContext) (schema.Activity, error) {
-			return turn.SendActivity(activity.MsgOptionText("Echo: " + turn.Activity.Text))
-		},
+func msTeamsMockMock(w http.ResponseWriter, r *http.Request) {
+	_, _ = w.Write([]byte("{\"id\":\"1\"}"))
+}
+
+type MockTokenValidator struct {
+}
+
+func (jv *MockTokenValidator) AuthenticateRequest(ctx context.Context, activity schema.Activity, authHeader string, credentials auth.CredentialProvider, channelService string) (auth.ClaimsIdentity, error) {
+	claims := map[string]interface{}{
+		"1": "1",
 	}
+	return auth.NewClaimIdentity(claims, true), nil
+}
```

Comment:
Can we please move this to a new file `bot_framework_adapter_fake.go`? So that other apps can import and use this?
Otherwise looks good.

Reply:
Done